### PR TITLE
Feature/remote configuration source

### PR DESCRIPTION
**Pull request by suriksarkisyan**

### CODE REVIEW

File: ios/qonversion_flutter.podspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ Pod::Spec.new do |s|
   s.source_files = 'Classes/**/*'
   s.dependency 'Flutter'
   s.platform = :ios, '9.0'
-  s.dependency "QonversionSandwich", "3.0.1"
+  s.dependency "QonversionSandwich", "3.1.0"
```
Comment (SpertsyanKM) [inline ios/qonversion_flutter.podspec:19]: Update in MacOS either

File: lib/src/dto/remote_configuration_source.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    unknownEnumValue: QRemoteConfigurationAssignmentType.unknown,
+  )
+  final QRemoteConfigurationAssignmentType assignmentType;
+
+  const QRemoteConfigurationSource(
+      this.id,
+      this.name,
+      this.type,
+      this.assignmentType
+      );
```
Comment (SpertsyanKM) [inline lib/src/dto/remote_configuration_source.dart:34]:
```suggestion
  );
```

File: lib/src/dto/remote_config.dart [hunk 3]
```diff
@@ -14,9 +15,13 @@ class QRemoteConfig {
   @JsonKey(name: 'experiment')
   final QExperiment? experiment;
 
+  @JsonKey(name: 'source')
+  final QRemoteConfigurationSource source;
```
Comment (SpertsyanKM) [inline lib/src/dto/remote_config.dart:19]: Doublecheck - neither this field nor it's nested fields are nullable, right?

Comment (suriksarkisyan) [reply lib/src/dto/remote_config.dart:19]: Right

**Review by SpertsyanKM — APPROVED** [review]

**Pull request merged**